core/src/main/java/org/apache/iceberg/MetricsConfig.java

```diff
@@ -74,6 +75,14 @@ public static MetricsConfig fromProperties(Map<String, String> props) {
     return spec;
   }
 
+  public void validateProperties(Schema schema) {
```

Comment:
I am not sure `validateProperties` is the most descriptive name in this case.

Something like `validateReferencedColumns` or similar would probably convey the purpose more. For example, we don't validate the property values here (if a value is invalid, it will be defaulted). We just check the config refers to valid columns.

Alternatively to changing the name, we can make this method static and accept both properties as well as schema. For example, we may invoke it as `MetricsConfig.validateProperties(props, schema)` and it will check both the column names as well as metrics modes (we can overload `fromProperties` so that it will throw an exception instead of defaulting).

Thoughts, @holdenk @rdblue @RussellSpitzer @shardulm94?

Reply:
I'd probably go for a static method and validating the value modes as well.

Reply:
I'm not sure about having this be a static method since it would end up constructing an instance of the object anyways, but I don't have strong feelings here. I lean towards giving it a name like `validateReferencedColumns` personally.

Reply:
Same for me. I don't feel strongly about this one.

Reply:
I think it is fine either way. I think `MetricsConfig#validateProperties` is reasonable, but I like `validateReferencedColumns` as well.